File: 준영/LV2/방문 길이.py
def solution(dirs):
    answer = 0

    dict = []

    now = [0, 0]
    for i in dirs:
        temp1 = now[:]
        if i == 'U':
            now[1] += 1
            if now[1] > 5:
                now[1] -= 1
                continue
        elif i == 'D':
            now[1] -= 1
            if now[1] < -5:
                now[1] += 1
                continue
        elif i == 'L':
            now[0] -= 1
            if now[0] < -5:
                now[0] += 1
                continue
        elif i == 'R':
            now[0] += 1
            if now[0] > 5:
                now[0] -= 1
                continue
        temp2 = now[:]
        path = str(temp1[0])+str(temp1[1])+str(temp2[0])+str(temp2[1])
        path2 = str(temp2[0])+str(temp2[1])+str(temp1[0])+str(temp1[1])

        if path not in dict and path2 not in dict:
            dict.append(path)
            dict.append(path2)
            answer += 1
        
    return answer

dirs = 'ULURRDLLU'
solution(dirs)

File: 준영/LV2/방문 길이.js
function solution(dirs) {
  var answer = 0
  const dict = []
  var now = [0, 0]

  for (let i = 0; i < dirs.length; i++) {
    const t = dirs[i];
    var temp1 = [...now]
    if (t === 'U') {
      now[1] += 1
      if (now[1] > 5) {
        now[1] -= 1
        continue
      }
    } else if(t === 'D'){
      now[1] -= 1
      if (now[1] < -5) {
        now[1] += 1
        continue
      }
    } else if(t === 'L'){
      now[0] -= 1
      if (now[0] < -5) {
        now[0] += 1
        continue
      }
    } else if(t === 'R'){
      now[0] += 1
      if (now[0] > 5) {
        now[0] -= 1
        continue
      }
    }
    var temp2 = [...now]
    var path1 = String(temp1) + String(temp2)
    var path2 = String(temp2) + String(temp1)

    if (!dict.includes(path1)&&!dict.includes(path2)) {
      dict.push(path1)
      dict.push(path2)
      answer += 1
    }

  }

  return answer
}

dirs = 'ULURRDLLU'
solution(dirs)